### PR TITLE
[Resource] catch foreing key deletion error

### PR DESCRIPTION
**Pull request by gperdomor**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | #4874
| License         | MIT

### CODE REVIEW

**Comment by michalmarcinkowski** [comment]
@gperdomor this is already implemented using [subscriber](https://github.com/michalmarcinkowski/Sylius/blob/26e41f3a9ec3bf608883210980c1ba73737c2d10/src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php#L75-L75).

**Comment by gperdomor** [comment]
Mmm... I get error for non html requests... We still need fix this, but in
that case, should be fixed in the listener... I will update the PR soon...

**Comment by michalmarcinkowski** [comment]
Exactly, you need to move this fix to the subscriber. Thanks!

**Comment by gperdomor** [comment]
@michalmarcinkowski updated, squashed and ready to review :D

**Comment by pjedrzejewski** [comment]
Looks good, although I am pretty sure we should not expose the sql exception in the API response.

**Comment by gperdomor** [comment]
@pjedrzejewski agree... We can use the translator to return a custom message.

Do you have a proposal for the message?

**Comment by pjedrzejewski** [comment]
@gperdomor I think we should reuse the translation that we use for html view. "entity is in use" or something like that. What do you think?

**Comment by gperdomor** [comment]
@pjedrzejewski updated!

File: src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php [hunk 1]
```diff
@@ -42,18 +45,26 @@ class ResourceDeleteSubscriber implements EventSubscriberInterface
     private $translator;
 
     /**
+     * @var RestViewHandler
+     */
+    protected $viewHandler;
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php]: This should be private and also an interface?

**Comment by gperdomor** [comment]
@pjedrzejewski ready

**Comment by pjedrzejewski** [comment]
Thanks @gperdomor!

**Pull request merged**